rmgpy/species.py

```diff
@@ -519,19 +520,22 @@ def copy(self, deep=False):
     def getAugmentedInChI(self):
         if self.aug_inchi is None:
             self.aug_inchi = self.generate_aug_inchi()
-            return self.aug_inchi
-        else:
-            return self.aug_inchi
+        return self.aug_inchi
 
     def generate_aug_inchi(self):
         candidates = []
         self.generateResonanceIsomers()
         for mol in self.molecule:
-            cand = mol.toAugmentedInChI()
-            candidates.append(cand)
-
-        candidates.sort()
-        return candidates[0] 
+            try:
+                cand = [mol.toAugmentedInChI(),mol]
+                candidates.append(cand)
```

Comment:
The `candidates.append(cand)` statement would ideally go into an `else` clause after the `except` clause.

Also, it would be best if you could identify the exception that's raised for the failure that you expect and specify it in the `except` statement.